firefox-plugins-installer

```diff
@@ -0,0 +1,152 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+DEST_DIR=${XDG_DATA_HOME}/firefox-plugins
+JAVA_DIR=${DEST_DIR}/java
+FLASH_DIR=${DEST_DIR}/flash
+PLUGINS_DIR=~/.mozilla/plugins
+WGET_COMMAND="wget -c --tries=5"
+CACHE_DIR=${XDG_CACHE_HOME}
+
+exit_with_error() {
+    echo $1
+    exit 1
+}
+
+JAVA_BUILD=8u112-b15
+JAVA_VERSION=`echo ${JAVA_BUILD} | cut -f 1 -d -`
+JAVA_FILENAME=jre-${JAVA_VERSION}-linux-x64.tar.gz
+JAVA_FILE_PATH=${CACHE_DIR}/${JAVA_FILENAME}
+JAVA_URL_FILENAME=http://download.oracle.com/otn-pub/java/jdk/${JAVA_BUILD}/${JAVA_FILENAME}
+JAVA_SHA256SUM=94053c6aa4d672b728c7788fb7d2676e5c6d7e7fcdbc1c86beaa796a083b4e5b
+JAVA_LINK=${PLUGINS_DIR}/libnpjp2.so
+
+# While the following would be a faster download of the current version,
+# previous versions are often removed from the server, so it is safest
+# for an automated script to use the stable version in the archive
+# http://fpdownload.macromedia.com/get/flashplayer/pdc/11.2.202.643/install_flash_player_11_linux.x86_64.tar.gz
+FLASH_VERSION=11.2.202.644
+FLASH_SHA256SUM=aba0e2f5940fc2588357e93f7bad84a0db51792b2f2bd1d24be9ef491e87c041
+FLASH_FILENAME=fp_${FLASH_VERSION}_archive.zip
+FLASH_FILE_PATH=${CACHE_DIR}/${FLASH_FILENAME}
+FLASH_URL_FILENAME=http://download.macromedia.com/pub/flashplayer/installers/archive/${FLASH_FILENAME}
+FLASH_LINK=${PLUGINS_DIR}/libflashplayer.so
+
+download_java() {
+    echo "Downloading ${JAVA_FILENAME}"
+    if ! ${WGET_COMMAND} --no-cookies --no-check-certificate --header "Cookie: oraclelicense=accept-securebackup-cookie" ${JAVA_URL_FILENAME} -O "${JAVA_FILE_PATH}"; then
+        exit_with_error "Failed to download ${JAVA_URL_FILENAME}"
+    fi
+
+    echo "Verifying ${JAVA_FILE_PATH}"
+    echo "${JAVA_SHA256SUM} ${JAVA_FILE_PATH}" | sha256sum -c > /dev/null 2>&1 || \
+    {
+        exit_with_error "sha256sum mismatch ${JAVA_FILE_PATH}"
+    }
+}
+
+install_java() {
+    rm -rf ${JAVA_DIR}
+    mkdir -p ${JAVA_DIR}
+
+    echo "Installing ${JAVA_FILE_PATH}"
+    if ! tar -xf ${JAVA_FILE_PATH} -C ${JAVA_DIR} ; then
+        exit_with_error "Cannot unpack tar file ${JAVA_FILENAME}"
+    fi
+
+    # Remove downloaded file
+    rm ${JAVA_FILE_PATH}
+}
+
+download_flash() {
+    echo "Downloading ${FLASH_FILENAME}"
+    if ! ${WGET_COMMAND} "${FLASH_URL_FILENAME}" -O "${CACHE_DIR}/${FLASH_FILENAME}"; then
+        exit_with_error "Failed to download ${FLASH_URL_FILENAME}"
+    fi
+
+    echo "Verifying ${FLASH_FILE_PATH}"
+    echo "${FLASH_SHA256SUM} ${FLASH_FILE_PATH}" | sha256sum -c > /dev/null 2>&1 || \
+    {
+        exit_with_error "sha256sum mismatch ${FLASH_FILE_PATH}"
+    }
+}
+
+install_flash() {
+    rm -rf ${FLASH_DIR}
+    mkdir -p ${FLASH_DIR}
+
+    echo "Installing ${FLASH_FILE_PATH}"
+
+    pushd ${CACHE_DIR}
```

Comment:
Yeah, this was copied from the old script that I had in eos-customer-service.  I'm not necessarily tied to using pushd -- it was just what seemed simplest to me when I originally wrote the code.  Feel free to change if desired, but for now I'll leave it alone.